src/fileset.rs

```diff
@@ -449,17 +453,56 @@ impl Fileset {
         self.filename_tokens.iter()
     }
 
+    pub fn entry_exists(&self, key: &str) -> bool {
+        // file exists
+        if self.exists(key) {
+            return true;
+        }
+
+        // directory lookup - check if there are any files within the directory
+        let dir = key.strip_prefix('/').unwrap_or(key);
+        let dirpath = Path::new(dir);
+
+        if self.directories.read().unwrap().contains(dirpath) {
+            return true;
+        }
+
+        match self.ordered_files.binary_search_by_key(&dirpath, |fe| fe.path.as_path()) {
+            // should be handled in `exists` already; something must be wrong
+            Ok(_) => unreachable!(),
+            Err(idx) => {
+                // there exists a file in the given directory
+                if self.ordered_files[idx].path.starts_with(&dirpath) {
+                    self.directories.write().unwrap().insert(dirpath.to_path_buf());
+                    return true;
+                }
+            }
+        }
+        false
+    }
+
+    pub fn verify_entry_exists(&self, entry: &str, token: &Token, max_sev: Severity) {
+        self.mark_used(entry);
+        if !self.entry_exists(entry) {
+            let msg = format!("file or directory {entry} does not exist");
+            report(ErrorKey::MissingFile, Item::File.severity().at_most(max_sev))
+                .msg(msg)
+                .loc(token)
+                .push();
+        }
+    }
+
     #[cfg(feature = "ck3")] // vic3 happens not to use
     pub fn verify_exists(&self, file: &Token) {
-        self.mark_used(&file.as_str().replace("//", "/"));
+        self.mark_used(file.as_str());
```

Comment:
Why this change? The stripping of `//` was done because paths containing that occur in the wild.